specification/search/Azure.Search/routes-service.tsp

```diff
@@ -0,0 +1,471 @@
+import "@azure-tools/typespec-azure-core";
+import "@typespec/rest";
+import "./models-service.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+
+namespace Search;
+
+interface DataSourcesOperations {
+    /**
+     * Creates a new datasource or updates a datasource if it already exists.
+     */
+    #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Pre-existing API contract"
+    @route("/datasources('{dataSourceName}')")
+    @put
+    createOrUpdate is SearchDataSourceOperation<
+        {
+            ...matchingHeader;
+            ...preferHeader;
+            /**
+             * The definition of the datasource to create or update.
+             */
+            @body
+            dataSource: SearchIndexerDataSource;
+        },
+        SearchIndexerDataSource
+    >;
+
+    /**
+     * Deletes a datasource.
+     */
+    #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Pre-existing API contract"
+    @route("/datasources('{dataSourceName}')")
+    @delete
+    delete is SearchDataSourceOperation<
+        {
+            ...matchingHeader;
+        },
+        void
+    >;
+
+    /**
+     * Retrieves a datasource definition.
+     */
+    #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Pre-existing API contract"
+    @route("/datasources('{dataSourceName}')")
+    @get
+    get is SearchDataSourceOperation<
+        {},
+        SearchIndexerDataSource
+    >;
+
+    /**
+     * Lists all datasources available for a search service.
+     */
+    #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Pre-existing API contract"
+    @route("/datasources")
+    @get
+    list is SearchOperation<
+        {
+            ...selectQuery;
+        },
```

Comment:
Just a tip, you dont have to change this, but if you only have one thing defined, like here you can just specify that directly. Example:
```suggestion
        selectQuery,
```